### PR TITLE
AJ-1666 Report PR images to sherlock

### PR DESCRIPTION
**Pull request by calypsomatic**
https://broadworkbench.atlassian.net/browse/AJ-1666
Before this PR:
The `publish-docker` workflow:
1. requires the input `input-tag`
2. is only called by the `tag` workflow
3. tags the new docker image with both the `input-tag` sent from `tag` and the github short hash.

The `tag` workflow, which is called manually for release:
1. discovers the new semver tag (e.g. 0.2.117)
2. passes the new tag to the `publish-docker` workflow
4. passes the new tag to sherlock to report the existence of the new docker image.

After this PR:
The `publish-docker` workflow:
1. does not require the input `input-tag`
2. is called both by the `tag` workflow and upon PR
3. If `input-tag` is present:
-  tags the new docker image with both the semver `input-tag` sent from `tag` and the github short hash
- reports to sherlock the existence of the new docker image with the semver tag

If `input-tag` is not present:
-  tags the new docker image only with the github short hash
- reports to sherlock the existence of the new docker image using the github short hash

The `tag` workflow:
1. discovers the new semver tag
2. passes the new tag to the `publish-docker` workflow
3. No longer calls report-to-sherlock, as this is now accomplished in `publish-docker`.

Reminder:

PRs merged into main will not automatically generate a PR in https://github.com/broadinstitute/terra-helmfile to update the WDS image deployed to kubernetes - this action will need to be triggered manually by running the following github action: https://github.com/DataBiosphere/terra-workspace-data-service/actions/workflows/tag.yml. Dont forget to provide a Jira Id when triggering the manual action, if no Jira ID is provided the action will not fully succeed. 

After you manually trigger the github action (and it completes with no errors), you must go to [the terra-helmfile](https://github.com/broadinstitute/terra-helmfile) repo and verify that this generated a PR that merged successfully.

The terra-helmfile PR merge will then generate a PR in [leonardo](https://github.com/DataBiosphere/leonardo).  This will automerge if all tests pass, but if jenkins tests fail it will not; be sure to watch it to ensure it merges. To trigger jenkins retest simply comment on PR with "jenkins retest".

### CODE REVIEW

**Comment by jladieu** [comment]
Thank you for the writeup describing the before/after state.  One question I still have: can you explain what the presence or absence of inputTag actually means?  Under what circumstances do we omit it or provide it, and who does the providing?

**Review by davidangb — APPROVED** [review]
did not test it myself, but looks great, thank you!

File: .github/workflows/publish-docker.yml [hunk 1]
```diff
@@ -38,6 +42,10 @@ jobs:
           echo $git_short_sha
           echo "git_short_sha=${git_short_sha}" >> $GITHUB_OUTPUT
 
+      - name: Determine new version
+        id: newVersion
+        run: echo "newVersion=${{ inputs.new-tag || steps.setHash.outputs.git_short_sha }}" >> "$GITHUB_OUTPUT"
```
Comment (jladieu) [inline .github/workflows/publish-docker.yml:47]: What does a value of `newVersion` look like?  I'm assuming it's something like `0.1.2wiopiwoipwo`.

Do we use that convention anywhere else?  I wonder if there should be a delimiter in between the tag & sha?

**Comment by calypsomatic** [comment]
> Thank you for the writeup describing the before/after state. One question I still have: can you explain what the presence or absence of inputTag actually means? Under what circumstances do we omit it or provide it, and who does the providing?

@jladieu The inputTag is the semvar generated upon calling the `tag` workflow for a release - e.g. 0.2.117 when previously we had 0.2.116.  So we only have this when we call the 'tag' workflow and are releasing a new version.  Whereas, on a PR, we haven't updated the version and have no new semvar to pass into the `publish-docker` workflow, leaving only the git commit hash to distinguish the new docker image.  Is that clearer?

**Review by jladieu — APPROVED** [review]

File: .github/workflows/publish-docker.yml (hunk 1 above)
Comment (calypsomatic) [reply .github/workflows/publish-docker.yml:47]: It's either the semvar - e.g. 0.2.117 - or the git hash - e.g. 4b57b53.  The docker images are always tagged with the git hash.  This `newVersion` variable is only used to report to sherlock what tags are present on docker images.  For releases we normally care about the semver, but for development and deploying to our BEEs, git hashes for our PRs are what we want to know about.  I can also rename this variable if that helps.

**Pull request merged**